test/extended/operators/daemon_set.go

```diff
@@ -0,0 +1,73 @@
+)
+
+var _ = g.Describe("[sig-arch] Managed cluster", func() {
+	oc := exutil.NewCLIWithoutNamespace("operator-daemonsets")
+
+	// Daemonsets shipped with the platform must be able to upgrade without disruption to workloads.
+	// Daemonsets that are in the data path must gracefully shutdown and redirect workload traffic, or
+	// mitigate outage by holding requests briefly (very briefly!). Daemonsets that are part of control
+	// plane (anything that changes state) must already be able to be upgraded without disruption by
+	// having any components that call them retry when unavailable up to a specific duration. Therefore
+	// all workloads must individually be graceful, and so a core daemonset can upgrade up to 10% of
+	// pods at a time. In the future, we may allow this percentage to be tuned via a global config, and
+	// this test would enforce that.
+	//
+	// Currently only applies to daemonsets that don't explicitly target the control plane.
+	g.It("should only include cluster daemonsets that have maxUnavailable update of 10 percent", func() {
+		// iterate over the references to find valid images
+		daemonSets, err := oc.KubeFramework().ClientSet.AppsV1().DaemonSets("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list daemonsets: %v", err)
+		}
+
+		// daemonsets that have an explicit exception, every entry here must have a bug associated
+		// e.g. "openshift-sdn/foo": "link to bug"
+		skipDaemonSets := map[string]string{
+			"openshift-multus/multus":                            "https://bugzilla.redhat.com/show_bug.cgi?id=1933159",
+			"openshift-sdn/sdn":                                  "https://bugzilla.redhat.com/show_bug.cgi?id=1933173",
+			"openshift-sdn/ovs":                                  "https://bugzilla.redhat.com/show_bug.cgi?id=1933174",
+			"openshift-cluster-csi-drivers":                      "https://bugzilla.redhat.com/show_bug.cgi?id=1933184",
+			"openshift-image-registry/node-ca":                   "https://bugzilla.redhat.com/show_bug.cgi?id=1933180",
+			"openshift-network-diagnostics/network-check-target": "https://bugzilla.redhat.com/show_bug.cgi?id=1933179",
+		}
+
+		var invalidDaemonSets []string
+		for _, ds := range daemonSets.Items {
+			if !strings.HasPrefix(ds.Namespace, "openshift-") {
+				continue
+			}
+			if ds.Spec.Selector != nil {
+				selector := labels.SelectorFromSet(ds.Spec.Template.Spec.NodeSelector)
+				if !selector.Empty() && selector.Matches(labels.Set(map[string]string{"node-role.kubernetes.io/master": ""})) {
+					continue
+				}
+			}
+			if bug, ok := skipDaemonSets[fmt.Sprintf("%s/%s", ds.Namespace, ds.Name)]; ok {
+				e2e.Logf("Skipping daemonset %s/%s from consideration: %s", ds.Namespace, ds.Name, bug)
+				continue
+			}
+			switch {
+			case ds.Spec.UpdateStrategy.RollingUpdate == nil,
+				ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable == nil:
+			case ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable.StrVal != "10%":
```

Comment:
@michaelgugino doesn't want to be bothered about tuning this for `machine-api-termination-handler`, so either this PR or any subsequent PR that bumps this check needs to allow 100%.

Reply:
MAO only has one daemonset, for spot instances watching for termination signals from the cloud provider.  In our case, we could be 100%, it makes no difference in impact to the end user or anything else.  Probably other components could be handled similarly, such as the node tuning bits.  So if we need to change this in the future, I'd prefer to just set 100% so we're never the slow point.

Reply:
ok, make it 100% and then i'll add the special case here (should be easy).  thanks for quick response.

Reply:
Actually, the one question about 100% is - do you think there is any value in partial degradation (i.e. if you totally botch it in an update, do you want to be all broken or max X percent broken)?

I think there's an analogy here to zone failures - after 33% percent or whatever we just say "we're not doing anything".  Should we make 33% our max so we can say "at least if we miss something 2/3 of the cluster should be working"?

Reply:
33% seems like a fine upper limit to me, and is probably acceptable in the general case for many components as well.

Reply:
Ok, I'll list 33% as an acceptable number

Reply:
33% vs. 10% doesn't seem like it would be all that significant.  Do we need to give components that flexibility?

Reply:
Any setting other than 'read setting from elsewhere' works for me.  10%, 33%, 100%, doesn't matter, as long as I don't need to have my operator manage that number back and forth.